### PR TITLE
HttpHeaders support (similar to docker client)

### PR DESCRIPTION
**Pull request by kpavel**
Signed-off-by: Pavel Kravchenko <kpavel@il.ibm.com>

Docker client already supports HttpHeaders from config file.
https://docs.docker.com/v1.7/docker/reference/commandline/cli/
Leveraging existing configuration file support in python docker client added support for generic HttpHeaders.

### CODE REVIEW

File: docker/auth/auth.py [hunk 1]
```diff
@@ -149,12 +149,19 @@ def load_config(config_path=None):
...
-                    return parse_auth(data)
-            log.debug("Couldn't find 'auths' section")
+                    if section == 'auths':
+                        log.debug("Found 'auths' section")
+                        res.update(parse_auth(data))
+                    elif section == 'HttpHeaders':
+                        log.debug("Found 'HttpHeaders' section!: " + str(section) + ':' + str(data))
+                        res.update({section: data})
+                if res:
+                    log.debug('Returning res: ' + str(res))
```
Comment (shin-) [inline docker/auth/auth.py]: This line seems unnecessary.

File: docker/auth/auth.py [hunk 2]
```diff
@@ -149,12 +149,19 @@ def load_config(config_path=None):
...
-                    log.debug("Found 'auths' section")
-                    return parse_auth(data)
-            log.debug("Couldn't find 'auths' section")
+                    if section == 'auths':
+                        log.debug("Found 'auths' section")
+                        res.update(parse_auth(data))
+                    elif section == 'HttpHeaders':
+                        log.debug("Found 'HttpHeaders' section!: " + str(section) + ':' + str(data))
+                        res.update({section: data})
+                if res:
```
Comment (shin-) [inline docker/auth/auth.py]: `return res` should happen regardless of its value.

File: docker/client.py [hunk 3]
```diff
@@ -84,6 +83,19 @@ def __init__(self, base_url=None, version=None,
                 )
             )
 
+    def update_headers(func):
```
Comment (shin-) [inline docker/client.py]: this should be moved to `decorators.py`

File: docker/client.py [hunk 4]
```diff
@@ -84,6 +83,19 @@ def __init__(self, base_url=None, version=None,
...
+        def inner(self, *args, **kwargs):
+            if 'HttpHeaders' in self._auth_configs:
+                if 'headers' not in kwargs:
+                    headers = self._auth_configs['HttpHeaders']
```
Comment (shin-) [inline docker/client.py]: Doesn't look like `headers` is being used anywhere.

**Comment by shin-** [comment]
Thanks for your contribution! 

I made a few comments, and the branch also needs to be rebased.

**Comment by kpavel** [comment]
Thanks for your review!
Fixed and rebased.

**Comment by shin-** [comment]
LGTM.

File: docker/auth/auth.py [hunk 5]
```diff
@@ -168,11 +167,17 @@ def load_config(config_path=None):
...
+            res = {}
             if data.get('auths'):
                 log.debug("Found 'auths' section")
-                return parse_auth(data['auths'])
+                res.update(parse_auth(data['auths']))
+            if data.get('HttpHeaders'):
+                log.debug("Found 'HttpHeaders' section")
+                res.update({'HttpHeaders': data['HttpHeaders']})
+            if res:
+                return res
```
Comment (dnephin) [inline docker/auth/auth.py]: I think this could use a new test case. I don't see anything asserting these are read properly.

**Comment by tombee** [comment]
Is this PR just missing a test to check HttpHeaders are read from the `~/.docker/config.json` file?

@kpavel Are you able to add this test?  Would be great to get this PR merged.

**Comment by tombee** [comment]
Awesome work @kpavel! :+1:

**Comment by shin-** [comment]
Closed via #899 - Thanks!

**Pull request closed without merging**